var/spack/repos/builtin/packages/py-tensorflow-probability/package.py

```diff
@@ -48,9 +59,32 @@ class PyTensorflowProbability(Package):
     depends_on("py-dm-tree", when="@0.12:", type=("build", "run"))
 
     # tensorflow_probability/python/__init__.py
-    depends_on("py-tensorflow@2.10:", when="@0.18:", type=("build", "run"))
-    depends_on("py-tensorflow@2.4:", when="@0.12:", type=("build", "run"))
-    depends_on("py-tensorflow@1.14:", when="@0.8:", type=("build", "run"))
+    # TODO: reactivate the JAX versions once the JAX package is available with newer versions
+    #  also add jaxlib as a dependency
+    # TODO: reactivate once TF 2.15 is ready https://github.com/spack/spack/pull/41069
+
+    variant("py-tensorflow", default=False, description="Build with TensorFlow support")
+    with when("+py-tensorflow"):
+        # depends_on("py-tensorflow@2.15", when="@0.23", type=("build", "run"))
+        depends_on("py-tensorflow@2.14:2", when="@0.22", type=("build", "run"))
+        depends_on("py-tensorflow@2.13:2", when="@0.21", type=("build", "run"))
+        depends_on("py-tensorflow@2.12:2", when="@0.20", type=("build", "run"))
+        depends_on("py-tensorflow@2.11:2", when="@0.19", type=("build", "run"))
+
+    # jaxlib is not required, as it's already a dependency of py-jax
+    variant("py-jax", default=False, description="Build with JAX support")
+    with when("+py-jax"):  # TODO: reactivate once the JAX package is available with newer versions
+        # depends_on("py-jax@0.4.20:0.4", when="@0.23", type=("build", "run"))
+        # depends_on("py-jax@0.4.16:0.4", when="@0.22", type=("build", "run"))
+        # depends_on("py-jax@0.4.14:0.4", when="@0.21", type=("build", "run"))
+        # depends_on("py-jax@0.4.8:0.4", when="@0.20", type=("build", "run"))
+        depends_on("py-jax@0.3.25:3", when="@0.19", type=("build", "run"))
+
+    depends_on(
+        "py-tensorflow@2.10:2", when="@0.18", type=("build", "run")
+    )  # keep here for backwards compatibility
+    depends_on("py-tensorflow@2.4:", when="@0.12:0.17", type=("build", "run"))
+    depends_on("py-tensorflow@1.14", when="@0.8:0.11", type=("build", "run"))
```

Comment:
TF 1.14 requires Python 3.7 and older, which Spack no longer supports. If it's not possible to build this version with newer TF, we can just remove this version.

Reply:
Removed, and deprecated the 2.4 (that is ~4 years old)